core/src/BitMatrix.cpp

```diff
@@ -23,16 +23,6 @@
 namespace ZXing {
 
 void
-BitMatrix::init(int width, int height)
```

Comment:
My reasoning for removal was that a) it is kind of a 'non-standard' interface for a simple vector/buffer class, b) it is not present in the Java code and c) it should be necessary pretty seldom, especially in situations where the function is logically 'creating' the buffer instead of modifying it. I think a functional approach without the `&outParam` pattern is generally more readable if applicable. I was thinking about suggesting something along the lines of a mix between `std::optional` and `DecodeStatus` as a return type to make the interfaces more natural (like the Java upstream) without reintroducing exceptions.

But you are right that my patch caused a lot more mallocs to happen in this particular example. Please see the added patch fixing this.